connectors/src/monitoring/index.ts

```diff
@@ -0,0 +1,45 @@
+export class MonitoringError extends Error {}
+import StatsDClient from "hot-shots";
+
+import mainlogger from "@connectors/logger/logger";
+const logger = mainlogger.child({ module: "temporal-monitoring" });
+
+export const statsDClient = new StatsDClient({});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const withMonitoring = <T extends Array<any>, U>(
+  fn: (...args: T) => U
+) => {
+  return async (...args: T): Promise<U> => {
+    let result: U | undefined = undefined;
+
+    try {
+      const startTime = new Date();
+      logger.info({ activity: fn.name }, `Activity started`);
+
+      const tags = [`activity:${fn.name}`];
+      statsDClient.increment("activities.count", 1, tags);
+
+      result = await fn(...args);
+      const endTime = new Date();
+      const elapsedMs = endTime.getTime() - startTime.getTime();
+      statsDClient.histogram(`activity.duration`, elapsedMs, tags);
+      logger.info(
+        { activity: fn.name, elapsedMs: elapsedMs },
+        `Activity ended successfully`
+      );
+      return result;
+    } catch (e) {
+      let tags = [`activity:${fn.name}`];
+      if (e instanceof MonitoringError) {
```

Comment:
I think it's much reacher f MonitoringError has a `type` and `message` that we explicitly set. We can then tag with the MonitoringError type and create bar graphs grouped by types instead of a big mass of "known_error"

I think this one is a desirable change.